src/native/libs/System.Security.Cryptography.Native/pal_ecc_import_export.c

```diff
@@ -592,3 +601,233 @@ EC_KEY* CryptoNative_EcKeyCreateByExplicitParameters(
+        goto error;
+
+    G = EC_POINT_new(group);
+
+    if (!G)
+        goto error;
+
+    if (!EC_POINT_oct2point(group, G, generatorBuffer, generatorBufferSize, NULL))
+        goto error;
+
+    if (!EcPointGetAffineCoordinates(group, *curveType, G, xBn, yBn))
+        goto error;
+
+    // Extract order (n)
+    if (!EVP_PKEY_get_bn_param(pkey, OSSL_PKEY_PARAM_EC_ORDER, &orderBn))
+        goto error;
+
+    // Extract cofactor (h)
+    if (!EVP_PKEY_get_bn_param(pkey, OSSL_PKEY_PARAM_EC_COFACTOR, &cofactorBn))
+        goto error;
+
+    // Extract seed (optional)
+    if (!seed || !EVP_PKEY_get_octet_string_param(pkey, OSSL_PKEY_PARAM_EC_SEED, NULL, 0, &sufficientSeedBufferSize))
+    {
+        *seed = NULL;
+        *cbSeed = 0;
+    }
+    else
+    {
+        seedBuffer = (unsigned char*)OPENSSL_malloc(sufficientSeedBufferSize);
+        size_t actualSeedSize;
+        if (EVP_PKEY_get_octet_string_param(pkey, OSSL_PKEY_PARAM_EC_SEED, seedBuffer, sufficientSeedBufferSize, &actualSeedSize))
+        {
+            *seed = seedBuffer;
+            *cbSeed = SizeTToInt32(actualSeedSize);
+        }
+        else
+        {
+            *seed = NULL;
+            *cbSeed = 0;
+        }
+    }
+
+    // Success; assign variables
+    *gx = xBn; *cbGx = BN_num_bytes(xBn);
+    *gy = yBn; *cbGy = BN_num_bytes(yBn);
+    *p = pBn; *cbP = BN_num_bytes(pBn);
+    *a = aBn; *cbA = BN_num_bytes(aBn);
+    *b = bBn; *cbB = BN_num_bytes(bBn);
+    *order = orderBn; *cbOrder = BN_num_bytes(orderBn);
+    *cofactor = cofactorBn; *cbCofactor = BN_num_bytes(cofactorBn);
+
+    rc = 1;
+    goto exit;
+
+error:
+    // Clear out variables from CryptoNative_EvpPKeyGetEcKeyParameters
+    if (*qx) BN_free((BIGNUM*)*qx);
+    if (*qy) BN_free(*qy);
+    if (d && *d) BN_free(*d);
+
+    *cbQx = *cbQy = 0;
+    *qx = *qy = NULL;
+    if (d) *d = NULL;
+    if (cbD) *cbD = 0;
+
+    // Clear our out variables
+    *curveType = Unspecified;
+    *cbP = *cbA = *cbB = *cbGx = *cbGy = *cbOrder = *cbCofactor = *cbSeed = 0;
+    *p = *a = *b = *gx = *gy = *order = *cofactor;
+    *seed = NULL;
+
+    if (xBn) BN_free(xBn);
+    if (yBn) BN_free(yBn);
+    if (pBn) BN_free(pBn);
+    if (aBn) BN_free(aBn);
+    if (bBn) BN_free(bBn);
+    if (orderBn) BN_free(orderBn);
+    if (cofactorBn) BN_free(cofactorBn);
+    if (seedBuffer) OPENSSL_free(seedBuffer);
+
+exit:
+    // Clear out temporary variables
+    if (group) EC_GROUP_free(group);
+    if (generatorBuffer) OPENSSL_free(generatorBuffer);
+    if (G) EC_POINT_free(G);
+
+    return rc;
+#else
+    (void)pkey;
+    (void)includePrivate;
+    *cbQx = *cbQy = 0;
+    *qx = *qy = NULL;
+    if (d) *d = NULL;
+    if (cbD) *cbD = 0;
+    *curveType = Unspecified;
+    *cbP = *cbA = *cbB = *cbGx = *cbGy = *cbOrder = *cbCofactor = *cbSeed = 0;
+    *seed = NULL;
+#endif
+}
+
+void CryptoNative_BufferFree(char* ptr)
+{
+    if (!ptr)
+    {
+        assert(false);
+        return;
+    }
+
+    free(ptr);
```

Comment:
Agree with Jeremy that it'd be easier to just shuffle the seed around as a BN.

However, if for whatever reason we don't do that, I think we are mixing allocators here. Should this be `OPENSSL_free`? It looks like this is used to free the `seed`, which is allocated with `OPENSSL_malloc`.